src/west/main.py

```diff
@@ -374,38 +373,44 @@ def set_zephyr_base(args):
         # At some point, we need a more flexible way to set environment
         # variables based on manifest contents, but this is good enough
         # to get started with and to ask for wider testing.
-        try:
-            manifest = Manifest.from_file()
-        except MalformedConfig as e:
-            log.die('Parsing of manifest file failed during command',
-                    args.command, ':', *e.args)
-        for project in manifest.projects:
```

Comment:
why was this fallback removed? it's the right default

Reply:
Because now `zephyr.base` is always written to `.west/config` so no need to try to detect it.

This further has the advantage that users can now decide to have a copy or side zephyr project next to the upstream zephyr and retarget `zephyr.base` in `.west/config`

As well as simplifying order of precedence.

For out-of-tree, it will be possible to point out a default `zephyr.base` in `~/.westconfig`

For the future when #215 goes in, it will also be possible to extend `CMake` to use `west config zephyr.base` if `ZEPHYR_BASE` is not set in environment, and thus allow user to invoke cmake in a west handled project without having to set `ZEPHYR_BASE`

Reply:
OK, I see now that the check was moved to `_update_west()`; sorry for missing that. But I still think the loop and setting of the config option belongs here; why is `_update_west()` a better place for it?

Reply:
To avoid `hidden` ways of fetching `zephyr.base` then we always write it during `west init`.
That means, a user can always fetch the value using `west config zephyr.base`, and each west managed project has it's own value.

If a user wants to change `zephyr.base`, as example for some local testing, he can then overwrite `zephyr.base` using environment setting, or do `west config zephyr.base <alternative_zephyr>`

As `zephyr.base` is now added during `west init` and must always be present, then the check you refer to is no longer needed.

Unfortunately, there are `west` managed projects that is not having this setting in their config file.
Thus, having the test in the `selfupdate` part is to ensure, that when user update to this `west` revision (or never) then this setting will be added to the conf file.

Reply:
Still confused. It seems like `set_zephyr_base()` can check if the config option is defined when trying to fall back on it, and loop over the projects to set it if undefined. That will keep commands like `west build` even if the user "manually" upgrades west with `git pull` for example. It won't break anything if the user manually sets ZEPHYR_BASE or runs `west config` to set `zephyr.base` either. And that can still be reverted later.

Reply:
yeah, as a final fallback, we could have it, I take your point, although I think it is only us who does manually updates `west`.
I will add an additional commit, similar to the `ConfigParser` fallback commit, then we can decide if changing it in the future.